semanticdb/semanticdb3/semanticdb3.md

```diff
@@ -141,129 +187,100 @@ It represents a location inside a document, such as a line inside a text file.
+the document where the definition is located.
 
-For example, this is how some of the definitions from the Scala standard library
-must be modelled:
-  * The `scala` package: `_root_.scala.`
-  * The `Int` class: `_root_.scala.Int#`
-  * The single-parameter `println` method: `_root_.scala.Predef.println(Ljava/lang/Object;)V.`
-  * The integer addition method: ``_root_.scala.Int#`+`(I)I.``.
+Global symbol format accommodates scoping rules of the underlying language,
+and is therefore language-dependent. For example, the `Int` class in
+the Scala standard library is modelled as `_root_.scala.Int#`, where
+the `_root_` prefix stands for the root package defined in the Scala Language
+Specification [\[17\]][17], and the pound sign (`#`) at the end of the symbol
+means that the symbol corresponds to a class as opposed to an object that
+would end with a dot (`.`). See [Languages](#languages) for more information.
 
 Global symbols must be unique across the universe of documents that
-a tool is working with at any given time. For example, if in such a universe,
-there exist multiple definitions of `Int` - e.g. coming from multiple different
-versions of Scala - then all references to those definitions will have
-`SymbolOccurrence.symbol` equal to the same `_root_.scala.Int#` symbol,
-and SemanticDB will not be able to provide information to disambiguate these references.
+a SemanticDB-based is working with at any given time. For example,
+if in such a universe, there exist multiple definitions of `Int` -
+e.g. coming from multiple different versions of Scala - then all references
+to those definitions will have `SymbolOccurrence.symbol` equal to the same
+`_root_.scala.Int#` symbol, and SemanticDB will not be able to provide
+information to distinguish these references from each other.
 
 In the future, we may extend SemanticDB to allow for multiple definitions
 that under current rules would correspond to the same global symbol.
-In the meanwhile, when global uniqueness is required, tool authors are advised to
-accompany global symbols with `SymbolInformation.location`.
+In the meanwhile, when global uniqueness is required, tool authors are advised
+to accompany global symbols with `SymbolInformation.location`.
 
 **Local symbols**. Correspond to a definition that isn't global (see above).
-Local symbol format is defined by the following two rules:
-  * Local symbols must start with `local`, so that they can be easily
-    distinguished from global symbols.
-  * Local symbols must be unique within the underlying document.
 
-For example, `x` in `def identity[T](x: T): T` may be modelled by local symbols
-`local0`, `local_x`, `local_identity_x`, as long as these names are unique within
-the underlying document. The same logic applies to the type parameter `T`,
-which is also a local definition.
+Local symbol format is language-agnostic and is a concatenation of `local`
+and a decimal number. For example, `x` in a Scala method
```

Comment:
we append non-decimal numbers to local symbols in the converter to the legacy api, I think "starts with local" is enough

Reply:
High-level API isn't governed by the spec though. Let's make this as prescriptive as possible.